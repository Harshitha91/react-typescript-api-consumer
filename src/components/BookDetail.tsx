import React from 'react';
import { useLocation } from 'react-router-dom';
import IBookData from '../types/BookDetailScreen';

const BookDetail: React.FC = () => {
  const location = useLocation();
  const book = location.state as IBookData;

  return (
    <div className="detail-page">
      <div className="col l6 padding-large">
        <h1 className="center">{book?.title}</h1>

        <h4 id="authors">Authors</h4>
        {book?.authors.length > 0 &&
          book?.authors.map(author => {
            return <p className="text-grey">{author.name}</p>;
          })}

        <h4 id="download_count">Download Count</h4>
        <p className="text-grey">{book?.download_count}</p>

        <h4 id="bookshelves">Bookshelves</h4>
        <p>
          {book?.bookshelves.length > 0 &&
            book.bookshelves.map(bookshelve => {
              return (
                <>
                  <span className="tag light-grey small margin-bottom">
                    {bookshelve}
                  </span>{' '}
                </>
              );
            })}
        </p>
        <br />

        <h4 id="subjects">Subjects</h4>
        {book?.subjects.length > 0 &&
          book.subjects.map(subject => {
            return <p className="text-grey">{subject}</p>;
          })}
        <br />
      </div>

      <div className="col l6 padding-large">
        <img
          src={book?.formats['image/jpeg']}
          className="round image opacity-min"
          alt="Menu"
        />
      </div>
    </div>
  );
};

export default BookDetail;
