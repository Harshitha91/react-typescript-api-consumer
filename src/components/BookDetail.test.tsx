import { shallow } from 'enzyme';
import * as React from 'react';
import BookDetail from './BookDetail';

//Mock for use location
const mockedUsedLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useLocation: () => mockedUsedLocation,
}));

describe('Book Detail Page', () => {
  it('render header', () => {
    expect(shallow(<BookDetail />).find('#authors').length).toEqual(1);
  });
  it('renders a email input', () => {
    expect(shallow(<BookDetail />).find('#download_count').length).toEqual(1);
  });
  it('renders a message input', () => {
    expect(shallow(<BookDetail />).find('#bookshelves').length).toEqual(1);
  });
  it('renders a submit button', () => {
    expect(shallow(<BookDetail />).find('#subjects').length).toEqual(1);
  });
  it('can render the Book Detail page', () => {
    const component = shallow(<BookDetail />);
    expect(component).toMatchSnapshot();
  });
});
