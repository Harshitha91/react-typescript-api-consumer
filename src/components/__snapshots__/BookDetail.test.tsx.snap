// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Detail Page can render the Book Detail page 1`] = `ShallowWrapper {}`;
